src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1140,6 +1230,123 @@ private void IssueBuildSubmissionToScheduler(BuildSubmission submission, bool al
             }
         }
 
+        private void ExecuteGraphBuildScheduler(BuildSubmission submission)
+        {
+            try
+            {
+                if (_shuttingDown)
+                {
+                    throw new BuildAbortedException();
+                }
+
+                var projectGraph = submission.BuildRequestData.ProjectGraph;
+                if (projectGraph == null)
+                {
+                    // TODO: Need to pass through other stuff? Look how RequestBuilder creates ProjectInstance
+                    projectGraph = new ProjectGraph(submission.BuildRequestData.ProjectGraphEntryPoints);
+                }
+
+                // TODO: Handle multiple entry points
```

Comment:
This is a tough one. `BuildSubmission` has a `BuildResult`, but in our case there could me multiple `BuildResults` for each entry point. Maybe for graph builds we just always aggregate into a root build result which doesn't map to a particular project...

Reply:
Yeah, that seems reasonable--is it legal, or do we need to construct a "traversal project" entrypoint?

Reply:
If VS wants to use this for design time builds, they'd want to retrieve the results for each node in the graph. We can do a design session on how to best represent this. We could subclass the BuildResult, multi purpose it (ew), maybe have a GraphBuildSubmission, etc